hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroReaderContext.java

```diff
@@ -138,8 +153,8 @@ public ClosableIterator<IndexedRecord> getFileRecordIterator(
     HoodieAvroFileReader reader = getOrCreateFileReader(storagePathInfo.getPath(), isLogFile, format -> {
       try {
         return (HoodieAvroFileReader) HoodieIOFactory.getIOFactory(storage)
-            .getReaderFactory(HoodieRecord.HoodieRecordType.AVRO).getFileReader(new HoodieConfig(),
-                storagePathInfo, format, Option.empty());
+            .getReaderFactory(HoodieRecord.HoodieRecordType.AVRO).getFileReader(hoodieReaderConfig,
+                storagePathInfo.getPath(), format, Option.empty());
```

Comment:
Use `storagePathInfo` directly to avoid fetching file size with an additional FS call.

Reply:
Fixed